doc/source/cluster/kubernetes.rst

```diff
@@ -1,271 +1,464 @@
-.. _ray-k8s-deploy:
-
+***********************
 Deploying on Kubernetes
-=======================
+***********************
+
+.. toctree::
+    :hidden:
+
+    /cluster/kubernetes-manual.rst
+    /cluster/kubernetes-gpu.rst
+
+
+Overview
+========
+The Ray Autoscaler uses the `NodeProvider interface`_ to manage Ray nodes.
+To autoscale in different contexts (e.g. on AWS, GCP, Azure, Kubernetes) the Autoscaler uses subclasses of this interface.
+The AWS, GCP, and Azure node providers provision one cloud instance per Ray node.
+
+The Kubernetes node provider on the other hand, spins up one Kubernetes pod per Ray node.
+Thus, when using the Ray Autoscaler with Kubernetes, each Ray node runs in its `own Kubernetes pod <https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/kubernetes/node_provider.py>`__.
```

Comment:
This is quite hard to parse -- 
1. Users don't need to know the existence of a Node Provider
2. Users will most care about "what is the K8s - Ray story and how do I get started"

```suggestion
```